python/seldon_core/metrics.py

```diff
@@ -1,9 +1,165 @@
+from prometheus_client.core import (
+    HistogramMetricFamily,
+    GaugeMetricFamily,
+    CounterMetricFamily,
+    CollectorRegistry,
+)
+from prometheus_client import exposition
+from prometheus_client.utils import floatToGoString
+
+from multiprocessing import Manager
+
+import numpy as np
+
 from typing import List, Dict
+import logging
+import json
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+ENV_SELDON_DEPLOYMENT_NAME = "SELDON_DEPLOYMENT_ID"
+ENV_MODEL_NAME = "PREDICTIVE_UNIT_ID"
+ENV_MODEL_IMAGE = "PREDICTIVE_UNIT_IMAGE"
+ENV_PREDICTOR_NAME = "PREDICTOR_ID"
+ENV_PREDICTOR_LABELS = "PREDICTOR_LABELS"
 
 COUNTER = "COUNTER"
 GAUGE = "GAUGE"
 TIMER = "TIMER"
 
+# This sets the bins spread logarithmically between 0.001 and 30
+BINS = [0] + list(np.logspace(-3, np.log10(30), 50)) + [np.inf]
+
+# Development placeholder
+LABELS = [
+    "worker_id",
+    "seldon_deployment_name",
+    "model_name",
+    "image_name",
+    "image_version",
+    "predictor_name",
+    "predictor_version",
+]
+
+image = os.environ.get(ENV_MODEL_IMAGE, "UNDEFINED:UNDEFINED")
+image_name, image_version = image.split(":")
+predictor_version = json.loads(os.environ.get(ENV_PREDICTOR_LABELS, "{}")).get(
+    "version", "UNDERFINED"
+)
+
+my_labels = {
+    "seldon_deployment_name": os.environ.get(ENV_SELDON_DEPLOYMENT_NAME, "UNDEFINED"),
+    "model_name": os.environ.get(ENV_MODEL_NAME, "UNDEFINED"),
+    "image_name": image_name,
```

Comment:
Probably best to add a name that makes it clear its not supported, ie NOT_IMPLEMENTED

Reply:
Oh ok I see you added it to model_name, but I also meant for image_name, image, version etc, as the names are not very intuitive that they are not implemented atm